NPE in BackupUtils

### Describe your problem!

Found on Google Play for current beta 2022.05.24-RC and bets 2022.05.13-RC:

```
java.lang.RuntimeException: 
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:3738)
  at android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:3905)
  at android.app.servertransaction.LaunchActivityItem.execute (LaunchActivityItem.java:103)
  at android.app.servertransaction.TransactionExecutor.executeCallbacks (TransactionExecutor.java:135)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:95)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2260)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loopOnce (Looper.java:210)
  at android.os.Looper.loop (Looper.java:299)
  at android.app.ActivityThread.main (ActivityThread.java:8166)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:556)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1037)
Caused by: java.lang.NullPointerException: 
  at java.io.Writer.<init> (Writer.java:88)
  at java.io.OutputStreamWriter.<init> (OutputStreamWriter.java:130)
  at cgeo.geocaching.utils.BackupUtils.createSettingsBackupInternal (BackupUtils.java:578)
  at cgeo.geocaching.utils.BackupUtils.backupInternal (BackupUtils.java:551)
  at cgeo.geocaching.utils.BackupUtils.backupStep2 (BackupUtils.java:379)
  at cgeo.geocaching.utils.BackupUtils.backup (BackupUtils.java:362)
  at cgeo.geocaching.MainActivity.onCreate (MainActivity.java:344)
  at android.app.Activity.performCreate (Activity.java:8178)
  at android.app.Activity.performCreate (Activity.java:8146)
  at android.app.Instrumentation.callActivityOnCreate (Instrumentation.java:1330)
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:3711)
```

### How to reproduce?

Unclear, maybe related to the new auto-backup

### Actual result after these steps?

_No response_

### Expected result after these steps?

_No response_

### Reproducible

Unclear

### c:geo Version

2022.05.24-RC

### System information

```text
53 occurrences, Android 10 and 12
```


### Additional Information

_No response_

--- a/main/src/cgeo/geocaching/utils/BackupUtils.java
+++ b/main/src/cgeo/geocaching/utils/BackupUtils.java
@@ -575,6 +575,9 @@ public class BackupUtils {
         try {
 
             final OutputStream os = ContentStorage.get().openForWrite(backupFile);
+            if (os == null) {
+                throw new IOException("Could not open backup file uri for writing:" + backupFile);
+            }
             writer = new OutputStreamWriter(os, StandardCharsets.UTF_8);
 
             final XmlSerializer xmlSerializer = Xml.newSerializer();

